controllers/complianceeventsapi/server.go

```diff
@@ -754,6 +754,26 @@ func getSingleComplianceEvent(db *sql.DB, w http.ResponseWriter,
 	}
 }
 
+// getPqErrKeyVals is a helper to add additional database error details to a log message. additionalKeyVals is provided
+// as a convenience so that the keys don't need to be explicitly set to interface{} types when using the
+// `getPqErrKeyVals(err, "key1", "val1")...“ syntax.
+func getPqErrKeyVals(err error, additionalKeyVals ...interface{}) []interface{} {
+	unwrappedErr := err
+
+	for unwrappedErr != nil {
```

Comment:
Why this is `for`?

Reply:
The idea is to keep unwrapping until either a pq.Error is found or we've unwrapped all that we could and thefore, nil is returned.

Reply:
oh I see so unwrap one by one then log db errors each